### PR TITLE
Transaction: Check that threshold is less than amount of signatures

### PR DESCRIPTION
**Pull request by eordano**

### CODE REVIEW

**Comment by eordano** [comment]
Closes #1080

File: lib/transaction/transaction.js [hunk 1]
```diff
@@ -447,6 +447,8 @@ Transaction.prototype._fromNonP2SH = function(utxo) {
 };
 
 Transaction.prototype._fromMultisigUtxo = function(utxo, pubkeys, threshold) {
+  $.checkArgument(threshold <= pubkeys.length,
```
Comment (braydonf) [inline lib/transaction/transaction.js]: Shouldn't this be "Threshold must be less than or equal to the number of publickeys"? If there are 5 pubkeys, the threshold should be 5 and lower, not greater (e.g. 1-of-5 signatures required or 5-of-5) This check may need to also be at: https://github.com/bitpay/bitcore/blob/master/lib/address.js#L278

Comment (maraoz) [inline lib/transaction/transaction.js]: Agreed with @braydonf on error message, and about the need of changing `address.js` to solve issue #1080.
To be sure, I'd add a test with @matiu's example, asserting it fails (see https://github.com/bitpay/bitcore/issues/1080)

File: test/transaction/transaction.js [hunk 2]
```diff
@@ -401,6 +401,17 @@ describe('Transaction', function() {
         transaction.addInput(new Transaction.Input(), Script.empty(), 0);
       }).to.not.throw();
     });
+    it('does not allow a threshold number greater than the amount of public keys', function() {
+      expect(function() {
+        transaction = new Transaction();
+        return transaction.from({
+          txId: '0000000000000000000000000000000000000000000000000000000000000000',
+          outputIndex: 0,
+          script: Script.buildMultisigOut([public1, public2], 2).toScriptHashOut(),
```
Comment (maraoz) [inline test/transaction/transaction.js]: shouldn't this work? 2 public keys, threshold of 2, isn't this a normal 2-of-2?

File: test/transaction/transaction.js [hunk 3]
```diff
@@ -401,6 +401,17 @@ describe('Transaction', function() {
...
+          satoshis: 10000
+        }, [], 1);
+      }).to.throw(errors.InvalidArgument);
```
Comment (maraoz) [inline test/transaction/transaction.js]: try to specify an error message to make this test more robust. i.e:
```
.to.throw('Threshold must be less than or equal to the number of publickeys');
```

File: test/transaction/transaction.js (hunk 2 above)
Comment (eordano) [inline test/transaction/transaction.js]: This should be a different test (the call to `from` is expected to fail, not the creation of the multisig)

File: lib/transaction/transaction.js (hunk 1 above)
Comment (eordano) [inline lib/transaction/transaction.js]: Fixed

File: test/transaction/transaction.js (hunk 2 above)
Comment (eordano) [inline test/transaction/transaction.js]: I changed that for an empty script to make it clearer

File: test/transaction/transaction.js (hunk 3 above)
Comment (eordano) [inline test/transaction/transaction.js]: Done

File: test/transaction/transaction.js (hunk 2 above)
Comment (maraoz) [inline test/transaction/transaction.js]: aaah, gotcha

**Comment by maraoz** [comment]
ACK, waiting for travis

**Pull request merged**